junit-runtime/src/main/scala/org/hamcrest/internal/ArrayIterator.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Ported from https://github.com/hamcrest/JavaHamcrest/
+ */
+package org.hamcrest.internal
+
+import java.{util => ju}
+
+class ArrayIterator(array: Array[AnyRef], private var currentIndex: Int = 0)
+    extends ju.Iterator[AnyRef] {
+
+  def this(array: AnyRef) = {
+    this(array match {
+        case arr: Array[AnyRef] => arr
```

Comment:
This is not correct. It will not match array of primitives. It should be `case arr: Array[_] =>`.